qutebrowser/keyinput/keyutils.py

```diff
@@ -246,7 +246,10 @@ def _parse_keystring(keystr):
                 yield '>'
                 assert not key, key
         elif c == '<':
-            special = True
+            if special:
```

Comment:
This seems to somewhat fix `<alt-less>` (I need to bind `<alt-shift-less>` to get this to work, I wonder why that needs `shift` but a plain `<` dosen't), but it dosen't fix `<alt-greater>` so it's not a full fix. 

I'm also not sure if this will change any error messages. I don't really like this whole parsing logic, it needs a proper grammar instead of consuming keys like this (in my opinion).

We could special-case `> ` right after `+` or `-`, but that feels really hacky to me.

Reply:
Can you help post what doesn't work with `<alt-greater>`? I've tested that and I don't see any problems.

I agree that a better parsing logic is better, I'll try my hands on it when I have more time to think about it.

> I wonder why that needs `shift` but a plain `<` dosen't

It's because even a plain `<` has a shift modifier on it, so the `str(KeyInfo)` call returns `'<'.upper()`.

Reply:
> Can you help post what doesn't work with `<alt-greater>`? I've tested that and I don't see any problems.

I get this error when trying to bind the key:

```
07:01:33 ERROR: Invalid value '<Alt+Shift+>>' - Could not parse '<Alt+Shift+>>': Got invalid key!
```

https://paste.the-compiler.org/view/0bd7f11a

> It's because even a plain `<` has a shift modifier on it, so the `str(KeyInfo)` call returns `'<'.upper()`.

We should probably fix that by adding a set of keys where shift dosen't matter, although that feels very hacky as well. `<alt-shift-less>` feels unintuitive and undiscoverable to me.

Reply:
> I get this error when trying to bind the key:

Thanks, I see the error too now.

> We should probably fix that by adding a set of keys where shift dosen't matter

I was thinking of the same thing, but I thought some other keyboard layout may need different modifiers to access `<`.

A quick look at https://en.wikipedia.org/wiki/Keyboard_layout shows there are a couple of obscure layouts where this is true.

Do you think it's worth implementing a full parser/pulling in a dependency with a grammar specification to do this?